### PR TITLE
Add getAccessList to ShareManager

### PR DESCRIPTION
**Pull request by rullzer**
Converting the getUsersSharingFile (https://github.com/nextcloud/server/blob/master/lib/public/Share.php#L91) to the share manager.

It now just returns an array. `[users => [], public => bool, remote=>bool]`.

There is not much fancy stuff around it.

@schiessle I think this is already enough for you?
@nickvergessen We then just write a simple helper function that for a given fileId gets the path.

Could you guys have a look?

### CODE REVIEW

**Comment by rullzer** [comment]
@nickvergessen @schiessle feedback please so the big cleanup can finally start ;)

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -1103,20 +1105,76 @@ public function userDeletedFromGroup($uid, $gid) {
...
 	 * [
-	 * 	'users' => ['user1', 'user2'],
-	 *  'groups' => ['group2']
+	 *  users => ['user1', 'user2', 'user4'],
+	 *  public => bool
+	 *  remote => bool
 	 * ]
 	 *
-	 * This is required for encryption
+	 * This is required for encryption/activities
```
Comment (nickvergessen) [inline lib/private/Share20/Manager.php]: activity is the appid 🙊

**Comment by nickvergessen** [comment]
So the current method only returns all the user ids?
Code looks good to me, but the helper function is crucial and needs to be done before removing anything ;)

**Review by nickvergessen — APPROVED** [review]

**Comment by rullzer** [comment]
@nickvergessen ok good. Probabaly best to have the helper function just in the activity app. Lets me clean this up a bit more and then make a PR there.

File: .htaccess [hunk 2]
```diff
@@ -72,3 +72,7 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
```
Comment (nickvergessen) [inline .htaccess]: 🤒

**Comment by rullzer** [comment]
@nickvergessen would https://github.com/nextcloud/server/pull/1341/commits/611f38ab1e4d839bb5eda9e2a2cefe68fb31439d be enough for you?

**Comment by nickvergessen** [comment]
I guess so, would have to test.

**Comment by rullzer** [comment]
Mmmm so as me @nickvergessen found out this isn't really failsave. Basically if a user receives a group share and they unshared it that user will still pop up in the list. Which is OK for encryption (since the user might gain access again) but not OK for notifiations.

I'll think about how to solve this properly.

**Comment by rullzer** [comment]
I'll pick this up soon. But close for now since it has conflicts anyway.

**Pull request closed without merging**